src/web/lib/components/PaginatedThemeSelector/index.js

```diff
@@ -20,6 +20,13 @@ export const PaginatedThemeSelector = ({
   images = []
 }) => {
   const itemCount = themes.length;
+
+  React.useEffect(() => {
+    if (itemCount && itemCount % perPage === 0 && currentPage) {
```

Comment:
Is this logic correct? If there is only one item per page (`perPage = 1`), then  `itemCount % perPage === 0` would always be true. As a result, whenever this logic is triggered, the current page would decrease by one, which doesn't seem right to me.

Did the condition here not work? https://github.com/mozilla/FirefoxColor/pull/890#discussion_r387768166

Reply:
it not did not work.

```javascript
if (itemCount && currentPage >= itemCount) {
  setCurrentPage(itemCount - 1);
}
```

if currentPage is 2 (page 3) but itemCount is 19, then it will never switch pages when itemCount gets changed to 18 (in this case we want to switch to currentPage to 1 (page 2) 

it seems that perPage can't be changed via the ui tho, right? so i wasn't accounting for perPage to ever be 1 (or different than 9).

Reply:
Ow facepalm. Need to multiply or divide by perPage to be able to meaningfully compare currentPage and itemCount.

Reply:
it seems that what I have works or I can slightly adjust what you did above to:

```javascript
 if (itemCount && currentPage >= (itemCount / perPage) ) {
    setCurrentPage(currentPage - 1);
 }
```

Reply:
That looks almost right, but only when either one theme is being deleted (which is often the case in practice).

If there was a delete-all feature, and the user is on the last page prior to deleting all themes, then the current page would be set to a non-existing page. To fix that, I believe that the logic should be:

```
let pageCount = Math.ceil(itemCount / perPage);
if (itemCount && currentPage >= pageCount) {
  setCurrentPage(pageCount - 1);
}
```